src/main/java/com/teragrep/jos_01/procfs/Task.java

```diff
@@ -0,0 +1,131 @@
+ * which are in The Program if any.
+ *
+ * Licensee must indemnify licensors and authors for any liability that these
+ * contractual assumptions impose on licensors and authors.
+ *
+ * To the extent this program is licensed as part of the Commercial versions of
+ * Teragrep, the applicable Commercial License may apply to this file if you as
+ * a licensee so wish it.
+ */
+package com.teragrep.jos_01.procfs;
+
+import com.teragrep.jos_01.procfs.status.ProcessStat;
+import com.teragrep.jos_01.procfs.status.Statm;
+
+import java.io.File;
+import java.util.ArrayList;
+
+public class Task {
+
+    private final long taskId;
+    private final long processId;
+    private final File procDirectory;
+
+    public Task(String taskId, Process parentProcess) {
+        this(Long.parseLong(taskId), parentProcess);
+    }
+
+    public Task(long taskId, Process parentProcess) {
+        this.taskId = taskId;
+        this.processId = parentProcess.pid();
+        this.procDirectory = new File("/proc/" + processId + "/task/", Long.toString(processId));
+    }
+
+    // Creates a Status object based on the chosen file name in /proc for this process. Overloaded to accept different kinds of Status objects
+    private ArrayList<String> reportStatistics(String procFileName) {
+        ArrayList<String> rows = readProcFile(procFileName);
+        return rows;
+    }
+
+    private ArrayList<String> readProcFile(String procFileName) {
+        RowFile procFile = new RowFile(procDirectory, procFileName);
+        ArrayList<String> rows = procFile.readFile();
+        return rows;
+    }
+
+    public ProcessStat stat() {
+        ArrayList<String> rows = readProcFile("stat");
+        return new ProcessStat(rows);
+    }
+
+    public Statm statm() {
+        ArrayList<String> rows = readProcFile("statm");
+        return new Statm(rows);
+    }
+
+    public ArrayList<String> availableProcFiles() {
+        ArrayList<String> fileNames = procFileNames(new ArrayList<String>(), procDirectory);
+        return fileNames;
+    }
+
+    private ArrayList<String> procFileNames(ArrayList<String> nameList, File file) {
+        if (!file.isDirectory()) {
+            nameList.add(file.getPath().replace(procDirectory.getPath(), ""));
+        }
+        else {
+            try {
+                for (File child : file.listFiles()) {
+                    procFileNames(nameList, child);
+                }
+            }
+            catch (NullPointerException npe) {
+                System.out.println("I/O Exception while attempting to access children of " + file.getPath());
+            }
+        }
+        return nameList;
+    }
+
+    public long tid() {
+        return taskId;
+    }
+
+    public long pid() {
+        return processId;
+    }
+
+    // Only the OS kernel can write or delete files from /proc, so if the process ID directory exists, the process is alive.
+    public boolean isAlive() {
+        if (procDirectory.exists()) {
```

Comment:
This could be just `return procDirectory.exists()`?

Reply:
Fixed